fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/connection/FHIRDbTenantDatasourceConnectionStrategy.java

```diff
@@ -0,0 +1,213 @@
+        final String datasourceName = datasourceBaseName + tenantId + "_" + dsId;
+        
+        // Note: we don't need any synchronization around ConcurrentHashMap, but that
+        // doesn't change the fact that we may look up the datasource and put it into
+        // the map more than once. That's fine. There aren't any integrity issues, just
+        // the chance of doing more work than necessary a single time, but we avoid the
+        // need for any synchronization. We cache locally, because we've found that JNDI
+        // lookups can become a bottleneck with high concurrency
+        datasource = datasourceMap.get(datasourceName);
+        if (datasource == null) {
+            // cache miss
+            try {
+                InitialContext ctxt = new InitialContext();
+                
+                datasource = (DataSource) ctxt.lookup(datasourceName);
+                datasourceMap.put(datasourceName, datasource);
+            } catch (Throwable e) {
+                // don't emit secrets in exceptions
+                FHIRPersistenceDBConnectException fx = new FHIRPersistenceDBConnectException("Failure acquiring datasource");
+                throw FHIRDbHelper.severe(log, fx, "Failure acquiring connection for datasource: " + datasourceName, e);
+            } finally {
+                if (log.isLoggable(Level.FINEST)) {
+                    log.exiting(CLASSNAME, METHODNAME);
+                }
+            }
+        }
+        
+        long start = System.nanoTime();
+        if (log.isLoggable(Level.FINE)) {
+            log.fine("Getting connection for tenantId/dsId: [" + tenantId + "/" + dsId + "]...");
+        }
+
+        // Now use the dsId/tenantId specific JEE datasource to get a connection
+        try {
+            connection = getConnection(datasource, tenantId, dsId);
+
+            if (log.isLoggable(Level.FINE)) {
+                long deltams = (System.nanoTime() - start) / NANOMS;
+                log.fine("Got the connection for [" + tenantId + "/" + dsId + "]. Took " + deltams + " ms");
+            }
+        } catch (Throwable e) {
+            // Don't emit secrets in case they are returned to a client
+            FHIRPersistenceDBConnectException fx =
+                    new FHIRPersistenceDBConnectException("Failure acquiring connection for datasource");
+            throw FHIRDbHelper.severe(log, fx, "Failure acquiring connection for datasource: " + datasourceName, e);
+        } finally {
+            if (log.isLoggable(Level.FINEST)) {
+                log.exiting(CLASSNAME, METHODNAME);
+            }
+        }
+        
+        return connection;
+    }
+    
+    @Override
+    public FHIRDbFlavor getFlavor() throws FHIRPersistenceDataAccessException {
+        return this.flavor;
+    }
+
+    /**
+     * Identify the flavor of the database using information from the
+     * datasource configuration.
+     * TODO duplicate code in FHIRDbConnectionStrategyBase. Refactor
```

Comment:
This is in Javadoc, I suggest a // after this comment or inside the method

Reply:
Moved comment and included reference to issue-916.